sql/analyzer/reorder_projections.go

```diff
@@ -207,27 +207,30 @@ func addIntermediateProjections(
 		deferredColumns = append(deferredColumns, findDeferredColumnsAndAliasReferences(s.Query)...)
 	}
 
-	if len(deferredColumns) > 0 {
-		schema := child.Schema()
-		var projections = make([]sql.Expression, 0, len(schema)+len(deferredColumns))
+	if len(deferredColumns) == 0 {
+		return child, same, err
+	}
+
+	schema := child.Schema()
+	var projections = make([]sql.Expression, 0, len(schema)+len(deferredColumns))
+
+	if rt, ok := child.(*plan.ResolvedTable); !(ok && plan.IsDualTable(rt)) {
```

Comment:
Do we still need this line? This looks like the only line with a behavior difference from the previous version and I suspect the direct child could be other things besides `ResolvedTable` (e.g. `IndexedTableAccess`). Check if all the tests pass without this line.